publisher-sdk/src/test/java/com/criteo/publisher/util/jsonadapter/ArrayListJsonAdapterTest.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ *    Copyright 2020 Criteo
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.criteo.publisher.util.jsonadapter
+
+import com.criteo.publisher.mock.MockedDependenciesRule
+import com.squareup.moshi.Moshi
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Rule
+import org.junit.Test
+import java.util.Arrays
+import javax.inject.Inject
+
+class ArrayListJsonAdapterTest {
+
+  @Rule
+  @JvmField
+  var mockedDependenciesRule = MockedDependenciesRule()
+
+  @Inject
+  private lateinit var moshi: Moshi
+
+  @Test
+  fun givenArraysArrayList_toJson_ShouldSerializeCorrectly() {
```

Comment:
Those classes are really technical and could deserve a better test coverage. Could you add more tests for those scenarios:
- what about deserialization ? for both ArrayList and Arrays.ArrayList ?
- what if there is null elements in the lists (a `List<T?>`)
- what if the list is null ?
- what if the list is empty ?
- what if the list contains a sublist ?
- what if the list as different kind of elements ? like a `List<Any>` such as `listOf(1, true, "foo")` ?